SnailWay/SnailWay.cs

```diff
@@ -8,18 +8,21 @@ public class SnailWay
     {
         public static int[] CalculateWay(int[][] grid)
         {
+            if(grid == null)
+            {
+                return null;
+            }
+
             List<int> snailWay = new List<int>();
-            var tempGrid = grid as IEnumerable<IEnumerable<int>>;
 
-            while (tempGrid.Count() != 0)
+            while (grid.Count() != 0)
             {
-                snailWay.AddRange(tempGrid.ElementAt(0));
-                tempGrid = tempGrid.Skip(1);
-                snailWay.AddRange((tempGrid.Select(line => line.Last())));
-                tempGrid = tempGrid.Select(line => line.SkipLast(1))
-                              .Reverse()
-                              .Select(line => line.Reverse());
+                snailWay.AddRange(grid[0]);
```

Comment:
Why AddRange for 1 element?

Reply:
Because it's impossible to add int[] to List<int>, if it was List<Enumerable<int>> that would be fine. Only AddRange() method allows adding several elements just once

Reply:
Correct, forgot that this is an array